ios/web_view/internal/cwv_web_view_extras.mm

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/web_view/public/cwv_web_view_extras.h"
+
+#include "base/apple/foundation_util.h"
+#include "base/json/json_writer.h"
+#include "base/strings/sys_string_conversions.h"
+#include "ios/web/common/user_agent.h"
+#include "ios/web/js_messaging/java_script_feature_manager.h"
+#include "ios/web/js_messaging/web_frame_internal.h"
+#include "ios/web/js_messaging/web_view_js_utils.h"
+#include "ios/web/public/js_messaging/web_frames_manager.h"
+#include "ios/web/web_state/ui/crw_web_controller.h"
+#include "ios/web/web_state/ui/wk_web_view_configuration_provider.h"
+#include "ios/web/web_state/web_state_impl.h"
+#include "ios/web_view/internal/cwv_web_view_configuration_internal.h"
+#include "ios/web_view/internal/cwv_web_view_internal.h"
+#include "ios/web_view/internal/web_view_browser_state.h"
+#include "ios/web_view/public/cwv_navigation_delegate.h"
+
+const CWVUserAgentType CWVUserAgentTypeNone =
+    static_cast<CWVUserAgentType>(web::UserAgentType::NONE);
+const CWVUserAgentType CWVUserAgentTypeAutomatic =
+    static_cast<CWVUserAgentType>(web::UserAgentType::AUTOMATIC);
+const CWVUserAgentType CWVUserAgentTypeMobile =
+    static_cast<CWVUserAgentType>(web::UserAgentType::MOBILE);
+const CWVUserAgentType CWVUserAgentTypeDesktop =
+    static_cast<CWVUserAgentType>(web::UserAgentType::DESKTOP);
+
+@implementation CWVWebView (Extras)
+
++ (BOOL)isRestoreDataValid:(NSData*)data {
+  return [self _isRestoreDataValid:data];
+}
+
+- (void)updateScripts {
+  // This runs `UpdateScripts` on the configuration provider which we will need
+  // to call in-place of `-[WKUserContentController removeAllUserScripts]` until
+  // all Brave JavaScript features are ported over to actual Chromium
+  // JavascriptFeature types and added to BraveWebClient
+  web::WKWebViewConfigurationProvider& config_provider =
+      web::WKWebViewConfigurationProvider::FromBrowserState(
+          self.webState->GetBrowserState());
+  config_provider.UpdateScripts();
+}
+
+- (void)createPDF:(void (^)(NSData* _Nullable))completionHandler {
```

Comment:
`createFullPagePdf` to match upstream method. It's much easier to find things when names match